abz/acousticbrainz.py

```diff
@@ -164,3 +164,185 @@ def process(path):
+                    if not itemuuid==None:
+                        itemuuid = itemuuid['itemuuid']
+                        tmpdict['recordings'].append(str(itemuuid))
+                        _update_progress(filepath, ":) Get UUID", GREEN)
+                        print()
+                    
+                    os.unlink(extractor_output)
+                    
+                if len(tmpdict['recordings'])>0:
+                    datasetdict['classes'].append(tmpdict)
+    
+    #print(datasetdict)
+    return datasetdict
+
+def _dataset_item_extractor(filepath):
+    """Extract lowlevel data from a dataset item
+    
+    Args:
+        filepath:   the fill path of the dataset item to process
+    Returns:
+        tmpname:    the full path of the generated lowlevel-data json file
+    """
+    fd, tmpname = tempfile.mkstemp(suffix='.json')
+    os.close(fd)
+    os.unlink(tmpname)
+    # avoid using default config file for datasets items MBID is not required
+    config.settings["profile_file"] = ""
+    try:
+        retcode, out = run_extractor(filepath, tmpname)
+        if retcode == 1:
+            print()
+            print(out)            
+            _update_progress(filepath, ":( extract", RED)
+            add_to_filelist(filepath, "extractor")
+        elif retcode < 0 or retcode > 0:
+            print()
+            print(out)
+            _update_progress(filepath, ":( unknown error", RED)
+        else:
+            if os.path.isfile(tmpname):
+                return tmpname
+            else:
+                return None
+    except KeyboardInterrupt:
+        print()
+        print("Action interrupted by the user!")
+        print("Stopped while extracting data from file:", filepath)
+        sys.exit(1)
+
+def _datasetdict_structure(dataset_name):
+    """Generates the skeleton for the dataset dictionary
+    
+    Args:
+        dataset_name:   the name of the dataset (the root folder of the dataset)
+    Returns:
+        datasetdict:    a dictionar ywith the dataset structure
+    """
+    datasetdict = {}
+    datasetdict['name'] = dataset_name
+    datasetdict['description'] = ''
+    datasetdict['public'] = 'true'
+    datasetdict['classes'] = []
+    return datasetdict
+
+def submit_dataset_item(filepath):
+    """Submit the lowlevel data of a dataset item and get the UUID from the server
+    
+    ARGS:
+        filepath: the path of the temp json file generated by the essentia extractor
+    Returns:
+        itemuuid: the item UUID retrieved/generated by the server
+    """
+    with open(filepath) as jsonfile:
+        try:
+            jsondata = json.load(jsonfile)
+        except (TypeError, ValueError):
+            print("Unknown Extractor Error")
+            _update_progress(filepath, ":( json format", RED)
+            return None
+    
+    host = config.settings["host"]
+    # This is a fake endpoints to be defined
+    url = compat.urlunparse(('http', host, '/datasetitem/', '', '',''))
```

Comment:
We should make this a configuration option, so that people (including us) don't need to edit the code of the client to change between http and https.
I propose that instead of having the host setting in the config file (https://github.com/MTG/acousticbrainz-client/blob/master/abz/default.conf#L2) we have a variable called `endpoint` or `base_endpoint`, which includes the scheme:
 `endpoint = http://localhost`
 `endpoint = https://acousticbrainz.org`
We can parse this setting to get the scheme and host. We should make sure that we still understand the existing `host` setting, and default the scheme to https if endpoint is not set.